src/sql/workbench/contrib/connection/browser/connectionStatus.ts

```diff
@@ -60,15 +65,48 @@ export class ConnectionStatusbarItem extends Disposable implements IWorkbenchCon
 	private _updateStatus(): void {
 		let activeConnection = TaskUtilities.getCurrentGlobalConnection(this._objectExplorerService, this._connectionManagementService, this._editorService, this._logService);
 		if (activeConnection) {
-			this._setConnectionText(activeConnection);
+			let uri = this._connectionManagementService.getConnectionUriFromId(activeConnection.id);
+			let info = this._connectionManagementService.getConnectionInfo(uri);
+			if (this._editorService.activeEditor) {
```

Comment:
Thoughts : 

1. I don't think we should ever have the connection info displayed and the the SPID value displayed be for different connections, which currently seems possible to have happen. Do we even want to display the status for anything if there isn't an active connected editor? It just seems confusing to have connection info but having no idea where it came from @cheenamalhotra @erinstellato-ms 
2. This should probably be moved out of the if check - updating this value seems like it should be completed separate from whether we have a global connection (which can be a number of different things)